cli/src/macro_opts.rs

```diff
@@ -0,0 +1,122 @@
+#[macro_export]
+macro_rules! shared_opts_wrapper {
```

Comment:
Given this struct applies the `#[derive(clap::Parsec)]` to the struct provided as input, I would call this macro `clap_parser_builder` (or `clap_parser_with_shared_opts_builder` if we want to be extra explicit)

On top of that, adding a small example as comment would help a lot understanding this macro
```rust
/// This macros builds a Clap argument parser by combining the provided structure
/// with common CLI options.
///
/// Available common CLI options:
/// - `config_dir` (add `config_dir: std::path::PathBuf`)
/// - `device` (add `device: Option<String>`)
/// - `addr` (add `addr: libparsec::ParsecAddr`)
/// - `token` (add `token: String`)
///
/// Example:
///
/// ```rust
/// crate::shared_opts_wrapper!(
///     #[with = config_dir]
///     pub struct ListUsers {
///         #[arg(short, long, default_value_t)]
///         skip_revoked: bool,
///     }
/// );
/// ```
///
/// Becomes:
///
/// ```rust
/// #[derive(clap::Parser)]
/// pub struct ListUsers {
///     #[arg(short, long, default_value_t)]
///     skip_revoked: bool,
///     #[doc = "Parsec config directory"]
///     #[arg(short, long, default_value_os_t = libparsec::get_default_config_dir(), env = $crate::utils::PARSEC_CONFIG_DIR)]
///     pub(crate) config_dir: std::path::PathBuf,
/// }
/// ```
```

Reply:
done